js-OOP/classes.js

```diff
@@ -0,0 +1,124 @@
+// Таск 3
+class ProductClass {
+    constructor(name, price, quantity) {
+      this.name = name;
+      this.price = price;
+      this.quantity = quantity;
+    }
+  
+    getTotalPrice() {
+      return this.price * this.quantity;
+    }
+  
+    purchase(quantity) {
+      if (this.quantity >= quantity) {
+        this.quantity -= quantity;
+        console.log(`Purchased ${quantity} ${this.name}(s)`);
+        console.log(`Remaining quantity: ${this.quantity}`);
+      } else {
+        console.log(`Not enough ${this.name}(s) in stock.`);
+      }
+    }
+  }
+  
+  const tShirt = new ProductClass("T-shirt", 20, 100);
+  const shoes = new ProductClass("Shoes", 50, 50);
+  const jeans = new ProductClass("Jeans", 30, 75);
+  
+  console.log(`Total price of T-shirt: $${tShirt.getTotalPrice()}`);
+  console.log(`Total price of Shoes: $${shoes.getTotalPrice()}`);
+  console.log(`Total price of Jeans: $${jeans.getTotalPrice()}`);
+  
+  tShirt.purchase(5);
+  shoes.purchase(60);
+  jeans.purchase(20);
+  
+  // Таск 4
+  class CarClass {
+    constructor(make, model, year) {
+      this.make = make;
+      this.model = model;
+      this.year = year;
+      this.speed = 0;
```

Comment:
speed property should be private
```suggestion
      this.#speed = 0;
```